seisbench/models/base.py

```diff
@@ -595,34 +683,67 @@ def annotate(
                                                This is usually acceptable for rotationally invariant models,
                                                e.g., most picking models.
         :type flexible_horizontal_components: bool
+        :param parallelism: If None, uses the `asyncio` implementation. Otherwise, defines the redundancy for each
+                            subjob, i.e., parallelism=2 would start each subjob twice. See the warning above for a
+                            discussion on parallelism for annotate.
+        :type parallelism: None, int
         :param kwargs:
         :return: Obspy stream of annotations
         """
-        nest_asyncio.apply()
-        call = self._annotate_async(
-            stream, strict, flexible_horizontal_components, **kwargs
+        # nest_asyncio.apply()
+        self._check_parallelism_annotate(stream, parallelism)
+
+        if parallelism is None:
+            nest_asyncio.apply()
+            call = self._annotate_async(
+                stream, strict, flexible_horizontal_components, **kwargs
+            )
+            return asyncio.run(call)
+        else:
+            return self._annotate_processes(
+                stream, strict, flexible_horizontal_components, **kwargs
+            )
```

Comment:
linked to the above comment, this is just showing that parallelism is allowed to be a value of 0, and in the control flow we then still use multiprocessing.